Issues/20 part 2

noahl opened the pull request:
Add JBoss facts to the default facts list, document JBoss facts, and
some minor cleanups of the JBoss scanning code.

Closes #20.

Commit:
Fixes for JBoss Scanning

Add JBoss facts to the default facts list, document JBoss facts, and
some minor cleanups of the JBoss scanning code.

Commit:
Merge branch 'master' into issues/20-part-2

chambridge commented on roles/brms/tasks/main.yml:
```diff
@@ -1,17 +1,17 @@
       raw: find / -name kie-api* 2> /dev/null | sed 's/.*kie-api-//g' | sed 's/.jar.*//g' | sort -u
-      register: kie-api-ver
+      register: jboss.brms.kie-api-ver
       ignore_errors: yes
```
Could we add a when condition for the jboss facts, so we only run commands when that fact is being requested?
```
when: '"jboss.brms.kie-api-ver" in facts_to_collect'
```
Better to not eat up the network with request if the fact isn't being requested.

noahl replied:
@chambridge done.

chambridge reviewed (commented):
LGTM. 
Minor comment to reduce unnecessary runs of tasks.

Commit:
Add when: clauses for JBoss roles

Add when: clauses so that Ansible will only run the JBoss rules when
their output is needed. Also standardize JBoss rules on '-' instead of
'_' in names, to make it easier to type names correctly.
